src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py

```diff
@@ -308,6 +309,13 @@ def load_annotations(self):
         for i, _ in enumerate(self.otx_dataset):
             class_indices = []
             item_labels = self.otx_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
+            if self.label_schema:
+                # NOTE: Parent labels might be missing in annotations.
+                # This code fills the gap just in case.
+                full_item_labels = set()
+                for label in item_labels:
+                    full_item_labels.update(self.label_schema.get_ancestors(label))
+                item_labels = full_item_labels
```

Comment:
What `label_schema.get_ancestors()` returns ? is it okay to overwrite the `item_labels` with the `full_item_labels` ?

Reply:
It returns [child, parent, grand parent, ...] which is needed for h-label classifier training & evaluation by definition. Geti gives such annotation, but Datumaro does not. So we need this patch.
In case of below hierarchy,
* human
  * male
    * boy
    * man 
  * female
    * girl
    * woman
We need labels like not only [boy], [woman], but also [male, man], [female, girl]
to train "male" vs "femail" high-level classifier.

So, yes, it should be added if absence.

Reply:
> It returns [child, parent, grand parent, ...] which is needed for h-label classifier training & evaluation by definition. Geti gives such annotation, but Datumaro does not. So we need this patch. In case of below hierarchy,
> 
> * human
>   
>   * male
>     
>     * boy
>     * man
>   * female
>     
>     * girl
>     * woman
> 
> We need labels like not only [boy], [woman], but also [male, man], [female, girl] to train "male" vs "femail" high-level classifier.
> 
> So, yes, it should be added if absence.

Thanks for the detailed reply.